cmd/retry.go

```diff
@@ -91,6 +91,8 @@ func newRetryTimer(unit time.Duration, cap time.Duration, jitter float64, doneCh
 	go func() {
 		defer close(attemptCh)
 		nextBackoff := 0
+		// Channel used to signal after the expiry of backoff wait seconds.
+		var backoffWait <-chan time.Time
```

Comment:
Don't use this you don't need this variable at all.  what you need to do is a simple behavior change. 

All you need is this 

```go
$ git diff cmd/retry.go
diff --git a/cmd/retry.go b/cmd/retry.go
index d3ad59e..6a5fe6b 100644
--- a/cmd/retry.go
+++ b/cmd/retry.go
@@ -91,16 +91,19 @@ func newRetryTimer(unit time.Duration, cap time.Duration, jitter float64, doneCh
        go func() {
                defer close(attemptCh)
                nextBackoff := 0
+               var timer *time.Timer
                for {
                        select {
                        // Attempts starts.
                        case attemptCh <- nextBackoff:
                                nextBackoff++
+                               timer = time.NewTimer(exponentialBackoffWait(nextBackoff))
                        case <-doneCh:
                                // Stop the routine.
+                               timer.Stop()
                                return
                        }
-                       time.Sleep(exponentialBackoffWait(nextBackoff))
+                       <-timer.C
                }
        }()
        return attemptCh
```